### PR TITLE
[KYUUBI #5780][AUTHZ] Treating PermanentViewMarker as LeafNode make code simple and got correct privilege object

### LINKED ISSUE #5780: [Improvement] Authz Treat PVM as leaf node to make it logic more simple
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Authz Treat PVM as leaf node to make it logic more simple

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
# :mag: Description
## Issue References 🔗

This pull request fixes #5780

## Describe Your Solution 🔧

Currently, we convert persist view to PermanentViewMaker,  but after optimizer, it changed its child, making it hard to do column prune and get the right column privilege object of persist view.

In this pr, we change PVM as a LeafNode, then we can directly treat it as a `HiveRelation` since we won't change its internal plan to make code simpler.
But we need to re-optimize the child plan after do privilege check.

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Behavior Without This Pull Request :coffin:
For sql such as  below table and view
```
CREATE TABLE IF NOT EXISTS $db1.$table1(id int, scope int);

CREATE TABLE IF NOT EXISTS $db1.$table2(
id int,
name string,
age int,
scope int)
.stripMargin);

CREATE VIEW $db1.$view1
AS
WITH temp AS (
    SELECT max(scope) max_scope
    FROM $db1.$table1)
SELECT id, name, max(scope) as max_scope, sum(age) sum_age
FROM $db1.$table2
WHERE scope in (SELECT max_scope FROM temp)
GROUP BY id, name
```
When we execute  query on `$db1.$view1`
```
SELECT id as new_id, name, max_scope FROM $db1.$view1
```

It will first execute the subquery in the query,  then got a un-correct column privilege

#### Behavior With This Pull Request :tada:
After this change, since PVM is a LeafNode, we won't execute the subquery under PVM, and we directly got the correct column privilege.

#### Related Unit Tests
Existed UT

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [x] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [x] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
@yaooqinn  WDYT about this solution?

**Review by bowenliang123 — APPROVED** [review]

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanentview/PermanentViewMarker.scala [hunk 1]
```diff
@@ -19,20 +19,15 @@ package org.apache.kyuubi.plugin.spark.authz.rule.permanentview
...
-    child: LogicalPlan,
-    catalogTable: CatalogTable,
-    outputColNames: Seq[String],
-    isSubqueryExpressionPlaceHolder: Boolean = false) extends UnaryNode
+case class PermanentViewMarker(child: LogicalPlan, catalogTable: CatalogTable) extends LeafNode
   with WithInternalChild {
 
   override def output: Seq[Attribute] = child.output
 
   override def withNewChildInternal(newChild: LogicalPlan): LogicalPlan =
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanentview/PermanentViewMarker.scala:31]: is this still necessary？

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanentview/PermanentViewMarker.scala]: Yea, inherit from `UnaryLike`. Can remove.

**Comment by AngersZhuuuu** [comment]
Any more suggestion?

**Comment by yaooqinn** [comment]
please take care of the tests

**Comment by AngersZhuuuu** [comment]
> please take care of the tests

Fixed

**Comment by AngersZhuuuu** [comment]
ping @yaooqinn GA passed and conflict resolved.

**Comment by yaooqinn** [comment]
The PR title is for refactoring, but actually what we do here is a bugfix, can we make it more precisely.

**Comment by AngersZhuuuu** [comment]
> The PR title is for refactoring, but actually what we do here is a bugfix, can we make it more precisely.

How about current, desc also full fill.

File: extensions/spark/kyuubi-spark-authz/src/main/resources/database_command_spec.json [hunk 2]
```diff
@@ -215,4 +215,4 @@
   } ],
   "opType" : "SWITCHDATABASE",
   "uriDescs" : [ ]
-} ]
+} ]
```
Comment (AngersZhuuuu) [inline extensions/spark/kyuubi-spark-authz/src/main/resources/database_command_spec.json:218]: I don't know why this pr needs to change this...But without this change, GA failed.

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/resources/database_command_spec.json:218]: cc @yaooqinn

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thank you, merged to master

**Pull request closed without merging**